### ISSUE TITLE
LabelMap creation from torchio.Image

### ISSUE THREAD

**Issue by romainVala** [fepegar/torchio#558]
**🐛Bug**
Sorry if I miss something here, but we used to declare LabelMap, from the torchio.Image function

**To reproduce**

```python
# Your code here
import torchio as tio
sujbad = tio.Image(type=tio.LABEL, path='/mypath/mask.nii.gz')
sujok  = tio.LabelMap(path='/mypath/mask.nii.gz')
```

**Expected behavior**

I expect those 2 declaration to be equivalent (since the type can be specify in the Image constructor)

**Actual behavior**
```python
type(sujbad)
```

```python-traceback
torchio.data.image.Image
```

**System info**

Output of `python <(curl -s https://raw.githubusercontent.com/fepegar/torchio/master/print_system.py)`:

```

Platform: Linux-4.15.0-139-generic-x86_64-with-glibc2.10
TorchIO:  0.18.39
PyTorch:  1.7.1
NumPy:    1.20.2
Python:   3.8.5 (default, Sep  4 2020, 07:30:14) 
[GCC 7.3.0]

```

**Comment by fepegar** [comment]
I agree this is not ideal. It comes from the old versions. Ideally, we would deprecate instantiating scalar images and label maps using `Image` and then just remove that possibility. Do you agree?

**Comment by romainVala** [comment]
yes it is confusion to use Image, with type argument, and this is not used ...
So to be sure I understand, you propose to never directly use Image ? (or is there case I miss ?)
and one should only go with label or scalar (sound ok, )

I will change my code, to this new way

**Comment by fepegar** [comment]
> So to be sure I understand, you propose to never directly use Image ? (or is there cases I miss ?)
and one should only go with label or scalar (it sounds ok to me )

Well, I think limiting to those two would be too restricting. For example, a probability map used for sampling isn't any of those so I would just maybe use `Image` with a custom type. But I haven't given this too much thought.

> I will change my code, to this new way

Thanks!

**Comment by romainVala** [comment]
I understand, that creating a torico.Image with type='label' is not a good idea, (because in other part of the code like for instance https://github.com/fepegar/torchio/blob/af86828360353dbe127d8195567ea4b9e83f238b/torchio/transforms/transform.py#L432, it will not be consider as a LabelMap

but is it also the case for intensity type :
should I also switch  
`tio.Image(type='intensity', path=path)   `
to  
`tio.ScalarImage(path=path)`

will it makes a difference somewhere ?

**Comment by fepegar** [comment]
> will it makes a difference somewhere ?

At the moment, I'm not sure. I would need to look into it. But please do move to `ScalarImage`, as in the examples and tutorials, as this is now considered the standard way and using `Image` for scalar images will be removed in the future.
